CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+# Contributing
+
+## Introduction
+First off, thank you for considering contributing to CIME. CIME is a community-driven
+project, so it's people like you that make CIME useful and successful.
+
+Following these guidelines helps to communicate that you respect the time of the
+developers managing and developing this open source project. In return, they
+should reciprocate that respect in addressing your issue, assessing changes, and
+helping you finalize your pull requests.
+
+We love contributions from community members, just like you! There are many ways
+to contribute, from writing tutorials or examples, improvements
+to the documentation, submitting bug report and feature requests, or even writing
+code which can be incorporated into CIME for everyone to use. If you get stuck at
+any point you can create an [issue on GitHub](https://github.com/ESMCI/CIME/issues).
+
+For more information on contributing to open source projects,
+[GitHub's own guide](https://guides.github.com/activities/contributing-to-open-source/)
+is a great starting point. Also, checkout the [Zen of Scientific Software Maintenance](https://jrleeman.github.io/ScientificSoftwareMaintenance/)
+for some guiding principles on how to create high quality scientific software contributions.
+
+## Getting Started
+
+Interested in helping extend CIME? Have code from your research that you believe others will
+find useful? Have a few minutes to tackle an issue? In this guide we will get you setup and
+integrated into contributing to CIME!
+
+## What Can I Do?
+* Tackle any unassigned [issues](https://github.com/ESMCI/CIME/issues) you wish!  
+
+* Contribute code you already have. It doesn’t need to be perfect! We will help you clean
+  things up, test it, etc.
+
+* Make a tutorial or example of how to do something.
+
+## How Can I Talk to You?
+Discussion of CIME development often happens in the issue tracker and in pull requests.
+
+## Ground Rules
+The goal is to maintain a diverse community that's pleasant for everyone.
+
+* Each pull request should consist of a logical collection of changes. You can
+  include multiple bug fixes in a single pull request, but they should be related.
+  For unrelated changes, please submit multiple pull requests.
+* Do not commit changes to files that are irrelevant to your feature or bugfix
+  (eg: .gitignore).
+* Be willing to accept criticism and work on improving your code; we don't want
```

Comment:
I would replace the following 

Be willing to accept criticism and work on improving your code; we don't want to break other users' code, so care must be taken not to introduce bugs.

with 

Be willing to accept constructive criticism as part of issuing a pull request, since the CIME developers are dedicated to ensuring that new features extend the system robustly and do not introduce new bugs.